### PR TITLE
NF-217 - first back-link implemented using Navigator

### PR DESCRIPTION
**Pull request by SteveSugden**

### CODE REVIEW

File: app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/controllers/makeclaim/ClaimTypeController.scala [hunk 1]
```diff
@@ -45,13 +46,13 @@ class ClaimTypeController @Inject() (
...
       val preparedForm = answers.claimType.fold(form)(form.fill)
-      Ok(claimTypePage(preparedForm))
+      Ok(claimTypePage(preparedForm, NavigatorBack(navigator.previousPage(ClaimTypePage, answers))))
     }
   }
 
   def onSubmit(): Action[AnyContent] = identify.async { implicit request =>
     form.bindFromRequest().fold(
-      formWithErrors => Future(BadRequest(claimTypePage(formWithErrors))),
+      formWithErrors => Future(BadRequest(claimTypePage(formWithErrors, NavigatorBack(None)))),
```
Comment (SteveSugden) [inline app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/controllers/makeclaim/ClaimTypeController.scala:55]: This needs re-factoring to get existing answers - either before or after the form validation fails

File: app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/navigation/Navigator.scala [hunk 2]
```diff
@@ -29,7 +29,6 @@ protected case class P(page: Page, destination: () => Call, canAccessGiven: User
 class Navigator @Inject() () extends Conditions with Ordering {
 
   private val pageOrder: Seq[P] = Seq(
-    P(FirstPage, routes.StartController.start, always),
     P(ClaimTypePage, makeclaim.routes.ClaimTypeController.onPageLoad, always),
```
Comment (SteveSugden) [inline app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/navigation/Navigator.scala:32]: Removing this "dummy" page which is only a marker  and only used by the StartController

File: app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/navigation/Navigator.scala [hunk 3]
```diff
@@ -51,7 +50,7 @@ class Navigator @Inject() () extends Conditions with Ordering {
 
   def nextPage(currentPage: Page, userAnswers: UserAnswers): Call =
     viewFor(pageOrder, nextPageFor(pageOrder, currentPage, userAnswers)).getOrElse(
-      throw new IllegalStateException(s"No page after $currentPage")
+      pageOrder.head.destination()
```
Comment (SteveSugden) [inline app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/navigation/Navigator.scala:53]: Bounce to first page in order if none found?   Means that the StartController continues to work without having to know what the first page is.

File: app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/views/makeclaim/ConfirmationPage.scala.html [hunk 4]
```diff
@@ -25,7 +26,7 @@
 
 @(reference: String)(implicit request: Request[_], messages: Messages)
 
-@layout(pageTitle = Title(messages("confirmation.title")), includeBackLink = false) {
+@layout(pageTitle = Title(messages("confirmation.title")), NavigatorBack(None)) {
```
Comment (SteveSugden) [inline app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/views/makeclaim/ConfirmationPage.scala.html:29]: When you know you don't want a back link

File: app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/views/Layout.scala.html [hunk 5]
```diff
@@ -33,22 +34,22 @@
         govukBackLink: GovukBackLink,
         appConfig: AppConfig
 )
-@(pageTitle: Title,
-        headBlock: Option[Html] = None,
-        includeBackLink: Boolean = true
+@(      pageTitle: Title,
+        backLink: NavigatorBack = NavigatorBack(None),  // TODO - remove default
```
Comment (crosslandwa) [inline app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/views/Layout.scala.html:38]: OBSERVATION: Can the TODO here be actioned in this PR (or is this a note for something to address later?)

Comment (SteveSugden) [reply app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/views/Layout.scala.html:38]: I remembered that TODO last night and then promptly forgot about it.   I think it's worth doing as the number of pages that don't need a back link is small and it removes the danger of forgetting to make a decision.

**Review by crosslandwa — APPROVED** [review]
Phew! Looks good

File: app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/views/makeclaim/ConfirmationPage.scala.html (hunk 4 above)
Comment (crosslandwa) [reply app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/views/makeclaim/ConfirmationPage.scala.html:29]: I like the explicitness of `NavigatorBack(None)` (over making that parameter optional and defaulting to a `None` internally)

File: app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/views/Layout.scala.html (hunk 5 above)
Comment (SteveSugden) [reply app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/views/Layout.scala.html]: Fixed!

**Review by crosslandwa — APPROVED** [review]

**Pull request merged**